lms/djangoapps/completion/api/v0/views.py

```diff
@@ -0,0 +1,70 @@
+""" API v0 views. """
+import logging
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from opaque_keys.edx.keys import CourseKey, UsageKey
+
+from lms.djangoapps.completion.models import BlockCompletion
+
+log = logging.getLogger(__name__)
+
+
+class CompletionBatchView(APIView):
+    permission_classes = []
+
+    def _validate(self, batch_object):
+        """
+        Performs validation on the batch object to make sure it is in the proper format.
+
+        Parameters:
+            batch_object: The object provided by the caller.
+
+        Raises:
+
+            KeyError:
+                If a key is not found in the object.
+
+            django.core.exceptions.ValidationError:
+                If a float is passed that is not between 0.0 and 1.0.
+        """
+        if 'user' not in batch_object:
+            raise KeyError("Key 'user' not found.")
+        if 'course_key' not in batch_object:
+            raise KeyError("Key 'course_key' not found.")
+        if 'blocks' not in batch_object:
+            raise KeyError("Key 'blocks' not found.")
+        for value in batch_object['blocks'].values():
+            if 1.0 < value < 0.0:
+                raise ValidationError("All block values must be between 1.0 and 0.0")
+        return
+
+    def post(self, request, *args, **kwargs):
+        """
+        Inserts a batch of completions.
+
+        REST Endpoint Format:
+        {
+          "user": "username",
+          "course_key": "course-key",
+          "blocks": {
+            "block_key1": 0.0,
+            "block_key2": 1.0,
+            "block_key3": 1.0,
+          }
+        }
+        """
+        batch_object = request.data or {}
+        self._validate(batch_object)
```

Comment:
This will currently bubble up the exception to the response.  Make sure we aren't sending a `500 Server error` (but rather a `400 Bad request`) response for user input errors.